Makefile

```diff
@@ -477,15 +484,28 @@ envtest: $(ENVTEST) ## Download envtest-setup locally if necessary.
 $(ENVTEST): $(LOCALBIN)
 	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
 
-.PHONY: bundle
-bundle: manifests kustomize set-image ## Generate bundle manifests and metadata, then validate generated files.
+.PHONY: set-image-ubi
+set-image-ubi: kustomize copy-config ## Set the controller UBI image
+	cd $(CONFIG_MANAGER_DIR) && $(KUSTOMIZE) edit set image controller=$(IMG_UBI)
+
+.PHONY: sdk-generate
+sdk-generate:
 	operator-sdk generate kustomize manifests -q
-	$(KUSTOMIZE) build $(KUSTOMIZE_BUILD_DIR)/manifests | operator-sdk generate bundle $(BUNDLE_GEN_FLAGS)
-	operator-sdk bundle validate ./bundle
+
+.PHONY: bundle
+bundle: manifests kustomize sdk-generate set-image-ubi ## Generate bundle manifests and metadata, then validate generated files.
+	@rm -rf $(BUNDLE_DIR)
+	@rm -f $(OPERATOR_BUILD_DIR)/bundle.Dockerfile
+	$(KUSTOMIZE) build $(CONFIG_BUILD_DIR)/manifests | operator-sdk generate bundle $(BUNDLE_GEN_FLAGS)
+	@$(COPYWRITE) headers &> /dev/null
+	@./hack/set_openshift_minimum_version.sh
+	@./hack/set_containerImage.sh
+	mv bundle.Dockerfile $(OPERATOR_BUILD_DIR)/bundle.Dockerfile
+	operator-sdk bundle validate $(BUNDLE_DIR)
```

Comment:
How does operator-sdk get installed, is that out of band? Not in this PR, but it might be nice to pin that to a commonly used version using the `tools/go.mod` and `make bootstrap` pattern that we have a few other places.

Reply:
Yeah good question, right now it is installed out of band. We should indeed add a `tools/go.mod` or another install script in `hack/` to pin a version.